Revision to leap seconds and calendar conventions (v2)

JonathanGregory opened the pull request:
**This PR is a work in progress and not yet ready for general comments**

Closed in favour of #612.

ChrisBarker-NOAA commented on ch04.adoc:
```diff
@@ -204,310 +220,236 @@ Optionally, a time coordinate variable may be indicated additionally by providin
-The difference was zero on that instant, when both calendars began.
-This means that a given datetime in the **`tai`** calendar represents an instant that is later than the same datetime in the **`utc`** calendar.
-See also <<leap-seconds>>.
+Moreover, a given datetime in the **`tai`** calendar represents an instant that is later than the same datetime in the **`utc`** calendar.
```
Honestly, I think we're putting more that necessary description in the CF doc -- if you don't know what TAI and UTC mean, you shouldn't be using them :-(.  -- but anyway, I think an example is in order -- examples are always more clear to users. Something like:

For example: `2025-11-13T21:00:43 TAI` is the same instant as: `2025-11-13T21:00:10 UTC`, as there have been 33 leap seconds introduced as of that date.

[Example taken from: http://www.csgnetwork.com/timetaidispcalc.html, which I presume is correct]

JonathanGregory replied:
I think you're commenting on the `tai` entry. Rather than adding an extra example, we can mirror what is said for `utc`, as follows (new and changed text in italics here in order to highlight it, but not in the document):

> For any given instant, the **`tai`** datetime is ahead of the **`utc`** datetime, where "ahead" means the same as it does when describing a timezone to the east as being ahead of one to the west. The difference between the two datetimes (**`tai`** minus **`utc`**) for a given instant of time is 10 seconds plus the net number of leap seconds introduced since _the instant with the datetime 1972-01-10 in the **`tai`** calendar, when the current definition of UTC came into force. This instant has the datetime 1972-01-01 00:00:00 in the **`utc`** calendar._ Moreover, a given datetime in the **`tai`** calendar represents an instant that is earlier than the same datetime in the **`utc`** calendar.

Thanks for the comment. I agree that if you're going to _write_ data in UTC or TAI you should already know what they mean. However, we can't depend on _users_ of the data being familiar with the detailed definitions. Yet they need to know them in order to interpret the data correctly.

JonathanGregory commented:
I've just corrected a mistake I made when adding Lars's table to the document. In this version, he had "yes" for "Leap days" in the non-model calendars. In order to give more information, I changed "yes" to "Julian" or "Gregorian",  but I forgot that standard is both, so I've now put "Julian/Gregorian" for its entry.
